js/coingecko.js

```diff
@@ -0,0 +1,219 @@
+                'createOrder': false,
+                'deposit': false,
+                'fetchBalance': false,
+                'fetchClosedOrders': false,
+                'fetchCurrencies': true,
+                'fetchDepositAddress': false,
+                'fetchMarkets': true,
+                'fetchMyTrades': false,
+                'fetchOHLCV': false,
+                'fetchOpenOrders': false,
+                'fetchOrder': false,
+                'fetchOrderBook': false,
+                'fetchOrders': false,
+                'fetchTicker': true,
+                'fetchTickers': true,
+                'fetchBidsAsks': false,
+                'fetchTrades': false,
+                'withdraw': false,
+            },
+            'urls': {
+                'logo': 'https://user-images.githubusercontent.com/2256515/43684626-2318304a-98d6-11e8-9b80-e46e9d248976.png',
+                'api': {
+                    'public': 'https://api.coingecko.com/api',
+                },
+                'www': 'https://coingecko.com',
+                'doc': 'https://www.coingecko.com/api/docs/v3',
+            },
+            'requiredCredentials': {
+                'apiKey': false,
+                'secret': false,
+            },
+            'api': {
+                'public': {
+                    'get': [
+                        'coins/',
+                        'coins/list',
+                        'coins/markets',
+                    ],
+                },
+            },
+            'vsCurrencies': [
+                'AED',
+                'ARS',
+                'AUD',
+                'BRL',
+                'BTC',
+                'CAD',
+                'CHF',
+                'CLP',
+                'CNY',
+                'CZK',
+                'DKK',
+                'ETH',
+                'EUR',
+                'GBP',
+                'HKD',
+                'HUF',
+                'IDR',
+                'ILS',
+                'INR',
+                'JPY',
+                'KRW',
+                'KWD',
+                'LKR',
+                'MXN',
+                'MYR',
+                'NOK',
+                'NZD',
+                'PHP',
+                'PKR',
+                'PLN',
+                'RUB',
+                'SAR',
+                'SEK',
+                'SGD',
+                'THB',
+                'TRY',
+                'TWD',
+                'USD',
+                'XAG',
+                'XAU',
+                'XDR',
+                'ZAR',
+            ],
+        });
+    }
+
+    async fetchMarkets () {
+        const coins = await this.publicGetCoinsList ();
+        let markets = [];
+        for (let i = 0; i < coins.length; i++) {
+            const coin = coins[i];
+            for (let j = 0; j < this.vsCurrencies.length; j++) {
+                markets.push ({
+                    'id': `${coin['id']}/${this.vsCurrencies[j].toLowerCase ()}`,
```

Comment:
Backticks are JS-specific, we should use basic string concatenation instead, otherwise this line will fail in Python and PHP.

Reply:
Thanks! Pushed a fix.